php/WP_CLI/FileCache.php

```diff
@@ -356,7 +357,10 @@ protected function prepare_write( $key ) {
 	 * @return string relative filename
 	 */
 	protected function validate_key( $key ) {
-		$url_parts = parse_url( $key );
+		$url_parts = function_exists( 'wp_parse_url' )
+			? wp_parse_url( $key )
+			// phpcs:ignore WordPress.WP.AlternativeFunctions.parse_url_parse_url -- Fallback.
+			: parse_url( $key );
```

Comment:
Just wondering based on the rest what's in this PR: shouldn't this block also be replaced by a call to `Utils\parse_url()` ?

Reply:
The problem is that it is not a URL, and the `Utils\parse_url()` (don't really know why)  adds `http://` to the variable of no scheme was found.

So, for a cache key, instead of returning `[ 'path' => 'foo' ]`, it returns `[ 'scheme' => 'http://', 'host' => 'foo' ]` instead.

Reply:
Just brainstorming here, what about adding a third, optional parameter to the `Utils\parse_url()` function which toggles whether `scheme` should be added ?
The parameter should then default to `true` to maintain BC.

Reply:
I initially added another argument first to fix this. Then I thought it might be overkill.

But now, as the `Utils\parse_url()` has a thousand code paths anyway and does multiple things, the extra argument seems much more sensible... :)